tests/unit/test_check.py

```diff
@@ -165,26 +280,83 @@ def _test_conflicts_parameters():
             ),
             id="multiple actions multiple problems",
         ),
+        pytest.param(
+            (
+                action_1 := factories.UpdateActionFactory(
+                    content_change=types_.ContentChange(base="a", server="b", local="b")
+                ),
+                action_2 := factories.UpdateActionFactory(
+                    content_change=types_.ContentChange(base="x", server="x", local="y")
+                ),
+            ),
+            False,
+            (
+                ExpectedProblem(
+                    path="/".join(action_1.path),
+                    description_contents=(
+                        "detected",
+                        "unmerged",
+                        "community contributions",
+                        constants.DISCOURSE_AHEAD_TAG,
+                        "/".join(action_1.path),
+                        "/".join(action_2.path),
+                    ),
+                ),
+            ),
+            id=(
+                "multiple actions one has base and local other has local and server diff, tag not "
+                "applied"
+            ),
+        ),
+        pytest.param(
+            (
+                action_1 := factories.UpdateActionFactory(
+                    content_change=types_.ContentChange(base="a", server="b", local="b")
+                ),
+                action_2 := factories.UpdateActionFactory(
+                    content_change=types_.ContentChange(base="x", server="x", local="y")
+                ),
+            ),
+            True,
+            (),
+            id=(
+                "multiple actions one has base and local other has local and server diff, tag "
+                "applied"
+            ),
+        ),
     ]
 
 
 @pytest.mark.parametrize(
-    "actions, expected_problems",
+    "actions, is_tagged, expected_problems",
     _test_conflicts_parameters(),
 )
 def test_conflicts(
     actions: tuple[types_.AnyAction, ...],
+    is_tagged: bool,
     expected_problems: tuple[ExpectedProblem],
     caplog: pytest.LogCaptureFixture,
+    repository_client: repository.Client,
 ):
     """
     arrange: given actions
     act: when conflicts is called with the actions
     assert: then the expected problems are yielded.
     """
     caplog.set_level(logging.INFO)
+    if is_tagged:
+        repository_client.tag_commit(
+            tag_name=constants.DISCOURSE_AHEAD_TAG, commit_sha=repository_client.current_commit
+        )
```

Comment:
I'm wondering if using an indirect fixture for this tag could be interesting

Reply:
Indirect fixture can be good if there is cleanup involved. In this case, it would add an additional layer of indirection which makes the test harder to follow